spec/reek/smells/nested_iterators_spec.rb

```diff
@@ -271,6 +271,15 @@ def bad(fred)
       expect(src).to reek_of(:NestedIterators, count: 2).with_config(config)
     end
   end
+
+  it 'should analyze safe-navigation-operator without error' do
+    src = '
+      def f
+        x&.tap {|_| }
+      end
+    '
+    expect(src).not_to reek_of(:NestedIterators, count: 2)
```

Comment:
I don't think this test makes much sense here. The error just happened there coincidentally. Having this test here is going to be super-confusing for us in 6 months.

How about:

* creating a directory in the [samples](https://github.com/troessner/reek/tree/master/spec/samples) directory called "super_smelly"
* put "inline.rb", "optparse.rb" and "redcloth.rb" in there
* create a new file called "syntax_checks.rb"
* Put the code above in it
* Update our [samples feature](https://github.com/troessner/reek/blob/master/features/samples.feature) to include this as well

WDYT? cc @mvz